packages/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,291 @@
+import React, { useState, useRef, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import Popover, { PopoverProps, Align, Justify } from '@leafygreen-ui/popover';
+import { useEventListener, useHandleEscape } from '@leafygreen-ui/hooks';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { transparentize } from 'polished';
+import debounce from 'lodash/debounce';
+import { trianglePosition } from './tooltipUtils';
+
+export const TriggerEvent = {
+  Hover: 'hover',
+  Click: 'click',
+} as const;
+
+type TriggerEvent = typeof TriggerEvent[keyof typeof TriggerEvent];
+
+export const Variant = {
+  Light: 'light',
+  Dark: 'dark',
+} as const;
+
+type Variant = typeof Variant[keyof typeof Variant];
+
+const baseStyles = css`
+  font-size: 14px;
+  line-height: 20px;
+  padding: 14px 16px;
+  border-radius: 3px;
+  box-shadow: 0px 2px 4px ${transparentize(0.85, uiColors.black)};
+`;
+
+const tooltipVariants: { readonly [K in Variant]: string } = {
+  [Variant.Dark]: css`
+    background-color: ${uiColors.gray.dark3};
+    color: ${uiColors.gray.light1};
+  `,
+
+  [Variant.Light]: css`
+    background-color: ${uiColors.gray.light3};
+    color: ${uiColors.gray.dark2};
+    border: 1px solid ${uiColors.gray.light2};
+  `,
+};
+
+const notchVariants = {
+  [Variant.Dark]: css`
+    background-color: ${uiColors.gray.dark3};
+    box-shadow: 0px 2px 4px ${transparentize(0.85, uiColors.black)};
+  `,
+  [Variant.Light]: css`
+    background-color: ${uiColors.gray.light3};
+    border: 1px solid ${uiColors.gray.light2};
+    box-shadow: 0px 2px 4px ${transparentize(0.85, uiColors.black)};
+  `,
+};
+
+interface TooltipProps
+  extends Omit<PopoverProps, 'active' | 'spacing' | 'refEl' | 'usePortal'> {
+  /**
+   * A slot for the element used to trigger the `Tooltip`.
+   */
+  trigger: React.ReactElement | Function;
```

Comment:
We should document what args the function is called with.

Reply:
👍